### ISSUE TITLE
Abort Local with docker kills the script but not the docker container

### ISSUE THREAD

**Issue by Horneth** [broadinstitute/cromwell#1126]
When running a task on docker locally, if the task is aborted, the script that feeds the wdl command into the docker container will be destroyed - but not the docker itself. 
Before abort:

```
wm0e9-683:cromwell tjeandet$ ps -ef | grep c72df249-ccf5-460b-8654-9606a655c669
1250915218 11002 10990   0  6:13PM ??         0:00.00 /bin/bash -c cat cromwell-executions/w/c72df249-ccf5-460b-8654-9606a655c669/call-t/script | docker run --rm -v /Users/tjeandet/Codebase/cromwell/cromwell-executions/w/c72df249-ccf5-460b-8654-9606a655c669/call-t:/root/w/c72df249-ccf5-460b-8654-9606a655c669/call-t -i ubuntu:latest /bin/bash <&0
1250915218 11004 11002   0  6:13PM ??         0:00.15 docker run --rm -v /Users/tjeandet/Codebase/cromwell/cromwell-executions/w/c72df249-ccf5-460b-8654-9606a655c669/call-t:/root/w/c72df249-ccf5-460b-8654-9606a655c669/call-t -i ubuntu:latest /bin/bash
1250915218 11015  2783   0  6:13PM ttys000    0:00.00 grep c72df249-ccf5-460b-8654-9606a655c669
```

After:

```
wm0e9-683:cromwell tjeandet$ ps -ef | grep c72df249-ccf5-460b-8654-9606a655c669
1250915218 11004     1   0  6:13PM ??         0:00.15 docker run --rm -v /Users/tjeandet/Codebase/cromwell/cromwell-executions/w/c72df249-ccf5-460b-8654-9606a655c669/call-t:/root/w/c72df249-ccf5-460b-8654-9606a655c669/call-t -i ubuntu:latest /bin/bash
1250915218 11021  2783   0  6:14PM ttys000    0:00.00 grep c72df249-ccf5-460b-8654-9606a655c669
```

**Comment by kshakir** [comment]
More info on the current issue:
- The local backend, running on the Config backend, supports running commands with or without docker.
- However the Config backend only allows `run-in-background` to be set for _both_ with and without docker.
- When `run-in-background` is set to true, the process id of the backgrounded process is stored for later killing.
- One cannot halt a docker container using [just `kill`](https://www.fpcomplete.com/blog/2016/10/docker-demons-pid1-orphans-zombies-signals). This gets ignored by the linux kernal and leaves the container running:
  > If you must send a SIGKILL to your process, use the docker kill command instead
- `docker kill` takes in the container id, not a pid.
- `docker run -d` outputs the container id, detaching the process.
- For this detached docker, one should set `run-in-background = false`, the opposite of Local without docker.

Workaround-- make the runtime attribute "docker" mandatory and run docker in detached mode:

```
Local {
  actor-factory = "cromwell.backend.impl.sfs.config.ConfigBackendLifecycleActorFactory"
  config {
    run-in-background = false

    runtime-attributes = """
    String docker
    String? docker_user
    """

    submit-docker = """
    docker run \
      -d --rm \
      ${"--user " + docker_user} \
      -v ${cwd}:${docker_cwd} \
      ${docker} \
      /bin/bash -c '/bin/bash ${script} > ${out} 2> ${err} < /dev/null'
    """

    kill = "docker kill ${job_id}"
    check-alive = "docker ps ${job_id}"
    job-id-regex = "([0-9a-f]+)"
  }
}
```

Actual fix-- Refactor the config backend, better separating the docker vs. non-docker config. Only `submit` vs. `submit-docker` are differentiated currently, but `run-in-background`, `job-id-regex`, `kill`, etc. are all different for docker vs. non-docker.

**TL;DR: The local backend should run the docker container in the background, capture the docker container id from stdout, and later issue a `docker kill <container id>`.**

**Comment by katevoss** [comment]
@kshakir do you have an idea of the effort for your Actual Fix: refactoring the config backend to separate out the Docker vs. Non-docker config?
